monai/data/meta_tensor.py

```
@@ -468,14 +502,29 @@ def affine(self) -> torch.Tensor:
+            self.spatial_ndim = _normalize_spatial_ndim(a.shape[-1] - 1, self.ndim)
+
+    @property
+    def spatial_ndim(self) -> int:
+        """Get the number of spatial dimensions."""
+        return getattr(self, "_spatial_ndim", 3)
+
+    @spatial_ndim.setter
+    def spatial_ndim(self, val: int) -> None:
+        """Set the number of spatial dimensions."""
+        if val < 1:
+            raise ValueError(f"spatial_ndim must be >= 1, got {val}")
```

Ruff (0.15.2)

[warning] Avoid specifying long messages outside the exception class

(TRY003)